packages/react-components/src/components/utils/filtrer-tree/sort.tsx

```diff
@@ -47,13 +47,26 @@ export const sortNodes = (nodes: TreeNode[], sortConfig: SortConfig[]): TreeNode
                 result = 0;
             } else {
                 if (typeof value1 === 'string' && typeof value2 === 'string') {
-                    const comp = (value1 as string).localeCompare(value2);
-                    result = (order === 'asc') ? -comp : comp;
+                    const number1 = parseValue(value1);
+                    const number2 = parseValue(value2);
+                    let comp = 0;
+                    if (number1 !== undefined && number2 !== undefined) {
+                        if (number1 < number2) {
+                            result = (order === 'asc') ? 1 : -1;
+                        } else if (number1 > number2) {
+                            result = (order === 'asc') ? -1 : 1;
+                        } else {
+                            result = 0;
+                        }
+                    } else {
+                        comp = (value1 as string).localeCompare(value2);
+                        result = (order === 'asc') ? -comp : comp;
+                    }
                 } else {
                     if (value1 < value2) {
```

Comment:
... and this code unreachable?

Reply:
you're right ... value1 and value2 are strings.